CHANGELOG.md

```diff
@@ -2,10 +2,349 @@
+* Fix recovering from FAAD2 decoding issues [#2850](https://github.com/mixxxdj/mixxx/pull/2850)
+
+### Audio Engine ###
+* Add support for Saved loops
+  [#2194](https://github.com/mixxxdj/mixxx/pull/2194)
+  [#3267](https://github.com/mixxxdj/mixxx/pull/3267)
+  [#3202](https://github.com/mixxxdj/mixxx/pull/3202)
+* Fix an issue when using multiple cue buttons at a time [#3382](https://github.com/mixxxdj/mixxx/pull/3382)
+* Fix synchronization of main cue point/position
+  [#4137](https://github.com/mixxxdj/mixxx/pull/4137)
+  [lp1937074](https://bugs.launchpad.net/mixxx/+bug/1937074)
+  [#4153](https://github.com/mixxxdj/mixxx/pull/4153)
+* Refactoring of beats object
+  [#4044](https://github.com/mixxxdj/mixxx/pull/4044)
+  [#4048](https://github.com/mixxxdj/mixxx/pull/4048)
+  [#4045](https://github.com/mixxxdj/mixxx/pull/4045)
+  [#4049](https://github.com/mixxxdj/mixxx/pull/4049)
+  [#4092](https://github.com/mixxxdj/mixxx/pull/4092)
+  [#4094](https://github.com/mixxxdj/mixxx/pull/4094)
+  [#4104](https://github.com/mixxxdj/mixxx/pull/4104)
+  [#4103](https://github.com/mixxxdj/mixxx/pull/4103)
+  [#4127](https://github.com/mixxxdj/mixxx/pull/4127)
+  [#4099](https://github.com/mixxxdj/mixxx/pull/4099)
+  [#4071](https://github.com/mixxxdj/mixxx/pull/4071)
+  [#4184](https://github.com/mixxxdj/mixxx/pull/4184)
+* Refactoring of audio engine code
+  [#2762](https://github.com/mixxxdj/mixxx/pull/2762)
+  [#2801](https://github.com/mixxxdj/mixxx/pull/2801)
+  [#2885](https://github.com/mixxxdj/mixxx/pull/2885)
+  [#2997](https://github.com/mixxxdj/mixxx/pull/2997)
+  [#3266](https://github.com/mixxxdj/mixxx/pull/3266)
+  [#4064](https://github.com/mixxxdj/mixxx/pull/4064)
+  [#4065](https://github.com/mixxxdj/mixxx/pull/4065)
+  [#4066](https://github.com/mixxxdj/mixxx/pull/4066)
+  [#4069](https://github.com/mixxxdj/mixxx/pull/4069)
+  [#4074](https://github.com/mixxxdj/mixxx/pull/4074)
+  [#4075](https://github.com/mixxxdj/mixxx/pull/4075)
+  [#4076](https://github.com/mixxxdj/mixxx/pull/4076)
+  [#4078](https://github.com/mixxxdj/mixxx/pull/4078)
+  [#4082](https://github.com/mixxxdj/mixxx/pull/4082)
+  [#4077](https://github.com/mixxxdj/mixxx/pull/4077)
+  [#4080](https://github.com/mixxxdj/mixxx/pull/4080)
+  [#4086](https://github.com/mixxxdj/mixxx/pull/4086)
+  [#4089](https://github.com/mixxxdj/mixxx/pull/4089)
+  [#4090](https://github.com/mixxxdj/mixxx/pull/4090)
+  [#4079](https://github.com/mixxxdj/mixxx/pull/4079)
+  [#4091](https://github.com/mixxxdj/mixxx/pull/4091)
+  [#4083](https://github.com/mixxxdj/mixxx/pull/4083)
+  [#4095](https://github.com/mixxxdj/mixxx/pull/4095)
+  [#4081](https://github.com/mixxxdj/mixxx/pull/4081)
+  [#4061](https://github.com/mixxxdj/mixxx/pull/4061)
+  [#4105](https://github.com/mixxxdj/mixxx/pull/4105)
+  [#4183](https://github.com/mixxxdj/mixxx/pull/4183)
+  [#4186](https://github.com/mixxxdj/mixxx/pull/4186)
+
+### Controller ###
+* ES6 based controller mapping system with module support
```

Comment:
```suggestion
* ES6 based controller mapping system
```
Using ES6 modules is still not possible as we're not loading the mappings as a module.

Reply:
For my understanding #2868 enabled the module support. Is it not yet usable?

Reply:
Anyway, I have improved the sentences to "Prepare code for upcoming ES6 based controller mapping system with module support" does it fit?

Reply:
>  For my understanding #2868 enabled the module support. Is it not yet usable?

Oh wow, I forgot about that PR entirely. I guess we technically do have module support, however, we have absolutely 0 docs or already written code that demonstrate this.

> Anyway, I have improved the sentences to "Prepare code for upcoming ES6 based controller mapping system with module support" does it fit?

The situation is too complex to describe in two sentences IMO so I don't care how we phrase it exactly. More important is that we document situation ASAP.

Reply:
There is no way to load ES6 modules currently.